ios/RNShopify.m

```diff
@@ -0,0 +1,320 @@
+        }
+
+        resolve(collectionDictionaries);
+    }];
+}
+
+RCT_EXPORT_METHOD(getProductTags:(NSUInteger)page resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getProductTagsPage:page completion:^(NSArray<NSString *> *tags, NSUInteger page, BOOL reachedEnd, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve(tags);
+    }];
+}
+
+RCT_EXPORT_METHOD(getProductsPage:(NSUInteger)page resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getProductsPage:page completion:^(NSArray<BUYProduct *> *products, NSUInteger page, BOOL reachedEnd, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve([self getDictionariesForProducts:products]);
+    }];
+}
+
+RCT_EXPORT_METHOD(getProductsWithTags:(NSUInteger)page tags:(NSArray<NSString *> *)tags resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getProductsByTags:tags page:page completion:^(NSArray<BUYProduct *> *products, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve([self getDictionariesForProducts:products]);
+    }];
+}
+
+RCT_EXPORT_METHOD(getProductsWithTagsForCollection:(NSUInteger)page collectionId:(nonnull NSNumber *)collectionId tags:(NSArray<NSString *> *)tags resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getProductsPage:page inCollection:collectionId withTags:tags sortOrder:BUYCollectionSortCollectionDefault completion:^(NSArray<BUYProduct *> *products, NSUInteger page, BOOL reachedEnd, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve([self getDictionariesForProducts:products]);
+    }];
+}
+
+RCT_EXPORT_METHOD(webCheckout:(NSArray *)cart resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject)
+{
+    _resolve = resolve;
+    _reject = reject;
+
+    BUYCheckout *checkout = [self createCheckoutFromCart:cart];
+
+    [self.client createCheckout:checkout completion:^(BUYCheckout *checkout, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        BUYWebCheckoutPaymentProvider *webPaymentProvider = [[BUYWebCheckoutPaymentProvider alloc] initWithClient:self.client];
+        webPaymentProvider.delegate = self;
```

Comment:
Regarding web checkout, we tried to complete the flow but we had an issue where we couldn't get the SDK to notify us when the user completed his order. We reported it to Shopify and they eventually gave us some guidance, but we decided to go with native checkout instead. I left the web part in the bridge as a reference to people that will want to use it. I also added a link to the issue we reported so people can find Shopify's response there. We can try to implement this ourselves in the future, but it depends on the time available. Or...we might wait for a contribution :).

Regarding the checkout instance, I thought about that in the past. Shopify's documentation says that you can get some other information from the completed checkout object such as order information. Their samples don't clear the checkout object. I think that there is no harm in keeping it and we might miss a use case if we clear it. A new checkout will overwrite it anyway.